src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_layer/shape_of.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ie_precision.hpp"
+#include "ngraph_functions/builders.hpp"
+#include <string>
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace ov::test;
+
+using ElementType = ov::element::Type_t;
+
+namespace GPULayerTestsDefinitions {
+typedef std::tuple<
+        InputShape,
+        ElementType                // Net precision
+> ShapeOfLayerGPUTestParamsSet;
+
+class ShapeOfLayerGPUTest : public testing::WithParamInterface<ShapeOfLayerGPUTestParamsSet>,
+                            virtual public SubgraphBaseTest {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<ShapeOfLayerGPUTestParamsSet> obj) {
+        InputShape inputShape;
+        ElementType netPrecision;
+        std::tie(inputShape, netPrecision) = obj.param;
+
+        std::ostringstream result;
+        result << "ShapeOfTest_";
+        result << std::to_string(obj.index) << "_";
+        result << "netPrec=" << netPrecision << "_";
+        result << "IS=";
+        result << CommonTestUtils::partialShape2str({inputShape.first}) << "_";
+        result << "TS=(";
+        for (const auto& shape : inputShape.second) {
+            result << CommonTestUtils::vec2str(shape) << "_";
+        }
+        result << ")";
+        return result.str();
+    }
+protected:
+    void SetUp() override {
+        targetDevice = CommonTestUtils::DEVICE_GPU;
+
+        auto netPrecision = ElementType::undefined;
+        InputShape inputShape;
+        std::tie(inputShape, netPrecision) = this->GetParam();
+
+        init_input_shapes({inputShape});
+
+        inType = ov::element::Type(netPrecision);
+        outType = ElementType::i32;
+
+        auto functionParams = builder::makeDynamicParams(inType, inputDynamicShapes);
+        auto paramOuts = helpers::convert2OutputVector(helpers::castOps2Nodes<opset3::Parameter>(functionParams));
+        auto shapeOfOp = std::make_shared<opset3::ShapeOf>(paramOuts[0], element::i32);
+
+        auto makeFunction = [](ParameterVector &params, const std::shared_ptr<Node> &lastNode) {
+            ResultVector results;
+
+            for (int i = 0; i < lastNode->get_output_size(); i++)
+                results.push_back(std::make_shared<opset1::Result>(lastNode->output(i)));
+
+            return std::make_shared<Function>(results, params, "ShapeOfLayerGPUTest");
+        };
+
+        function = makeFunction(functionParams, shapeOfOp);
+    }
+};
+
+TEST_P(ShapeOfLayerGPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+
+const std::vector<ElementType> netPrecisions = {
+        // ElementType::f32,
```

Comment:
How about deleting it if you don't use it?

Reply:
done